osbrain/tests/test_agent.py

```diff
@@ -50,7 +50,9 @@ def test_early_agent_proxy(nsaddr):
     # Locate agent now
     a0 = Proxy('a0', timeout=3.)
     # Just check agent is ready
-    assert a0.ping() == 'pong'
+    safe_status = a0._safe
+    assert a0.unsafe.ping() == 'pong'
+    a0._safe = safe_status
```

Comment:
Note that `._safe` is a variable that is set to `True`/`False` just after `.safe`/`.unsafe` but then, once the proxy executes one method, its value is set back to `._default_safe`, which is gotten from the environment variable or the parameter when the proxy is initially created.

In your example it would be something like:

```python
class C():
  def __init__(self):
    self.x = 3
    self.default_x = 3

  @property
  def set_four(self):
    self.x = 4
    return self

  def print_x(self):
    print(self.x)
    self.x = self.default_x

c = C()
c.print_x() # prints 3
c.set_four.print_x() # prints 4
c.print_x() # prints 3
```